plugins/de/opendiabetes/vault/plugin/importer/vaultodv/VaultODVImporter.java

```diff
@@ -0,0 +1,170 @@
+                    LOG.log(Level.SEVERE, "Missing hash algorithm for signature. No file exported!", ex);
+                    return false;
+                }
+                DigestInputStream digestInputStream = new DigestInputStream(zipInputStream, messageDigest);
+
+                ZipEntry tmpEntry;
+                String signature = null;
+                boolean processingResult = false;
+                while ((tmpEntry = zipInputStream.getNextEntry()) != null) {
+                    switch (tmpEntry.getName()) {
+                        case VaultODVExporter.DATA_ZIP_ENTRY:
+                            // process data while creating signature
+                            processingResult = super.processImport(digestInputStream, importPath);
+                            break;
+                        case VaultODVExporter.SIGNATURE_ZIP_ENTRY:
+                            // read signature as string
+                            signature = new BufferedReader(new InputStreamReader(zipInputStream))
+                                    .lines().collect(Collectors.joining("\n"));
+                            break;
+                        default:
+                            LOG.warning("Found unexpected entry: " + tmpEntry.getName());
+                            break;
+                    }
+                }
+                zipInputStream.close();
+
+                if (processingResult) { // if not, data entry is missing
+                    // check signature
+                    String sigString = (new HexBinaryAdapter()).marshal(messageDigest.digest());
+                    if (sigString.equalsIgnoreCase(signature)) {
+                        return true;
+                    } else {
+                        LOG.severe("Signature check failed! File will be dropped.");
+                        return false;
+                    }
+                } else {
+                    return false;
+                }
+            } catch (FileNotFoundException ex) {
+                LOG.log(Level.SEVERE, "Error opening a FileInputStream for file: "
+                        + importPath, ex);
+                return false;
+            } catch (IOException ex) {
+                LOG.log(Level.SEVERE, "Error reading compressed file: "
+                        + importPath, ex);
+                return false;
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean loadConfiguration(final String filePath) {
+            LOG.log(Level.WARNING, "VaultODVImporter does not support configuration.");
+            return false;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        protected List<VaultEntry> parseEntry(final CsvReader csvReader) throws UnsupportedOperationException {
+            throw new UnsupportedOperationException("ParseEntry not implemented for VaultODVImporter plugin! (Not needed");
+        }
+
```

Comment:
Would write real JavaDoc to explain why it is implemented like this, not inherit it.